File: src/main/java/com/team573/gongguri/domain/groupPurchase/entity/GroupPurchaseParticipant.java
package com.team573.gongguri.domain.groupPurchase.entity;

import com.team573.gongguri.domain.member.entity.Member;
import com.team573.gongguri.global.entity.BaseEntity;
import jakarta.persistence.*;
import lombok.Getter;
import lombok.NoArgsConstructor;


@Entity
@Getter
@NoArgsConstructor
@Table(name = "group_purchase_participant")
public class GroupPurchaseParticipant extends BaseEntity {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long groupParticipantId;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "group_id", nullable = false)
    private GroupPurchase groupPurchase;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "member_id", nullable = false)
    private Member member;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private ParticipationStatus participationStatus;
}

File: src/main/java/com/team573/gongguri/domain/grouppurchase/entity/GroupPurchaseParticipant.kt
package com.team573.gongguri.domain.grouppurchase.entity

import com.team573.gongguri.domain.member.entity.Member
import com.team573.gongguri.global.entity.BaseEntity
import jakarta.persistence.*

@Entity
@Table(name = "group_purchase_participant")
class GroupPurchaseParticipant (
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val groupParticipantId: Long? = null,

    var deposit:Boolean = false,

    @JoinColumn(name = "group_id", nullable = false)
    @ManyToOne(fetch = FetchType.LAZY)
    var groupPurchase: GroupPurchase,

    @JoinColumn(name = "member_id", nullable = false)
    @ManyToOne(fetch = FetchType.LAZY)
    var member: Member,

    @Column(nullable = false)
    @Enumerated(EnumType.STRING)
    var participationStatus: ParticipationStatus
) : BaseEntity() {

    fun confirmDeposit() {
        this.deposit = true
    }

    fun cancelMember() {
        this.participationStatus = ParticipationStatus.CANCELLED
    }
}
